File: Examples/Regression/Simple_Linear_Regression/myFirstSimpleLinearRegression.py
#Simple Linear Regression
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

datasets = pd.read_csv('Salary_Data.csv')
print(datasets)

#taking independent and depended variables from dataset and convert into matrixs
X = datasets.iloc[:, :-1].values
Y = datasets.iloc[:, 1].values

#Splitting dataset into train and test.
from sklearn.model_selection import train_test_split
X_train,X_test, Y_train, Y_test = train_test_split(X, Y, test_size=1/3, random_state=0)

#Fitting Simple Linear Regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

#Predict the Test set result
Y_pred = regressor.predict(X_test)

#Visualising the training set results
plt.scatter(X_train, Y_train, color='red')
plt.plot(X_train, regressor.predict(X_train))
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#Visualising the testing set results
plt.scatter(X_test, Y_test, color='red')
plt.plot(X_train, regressor.predict(X_train))
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: Examples/Regression/Simple_Linear_Regression/myFirstSimpleLinearRegression.R
#Simple Linear regression Model

#importing Datasets
dataset = read.csv("Salary_Data.csv")

#Splitting the dataset into the training set and test set
#install.packages('caTools')
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Fitting Simple Linear Regression to the training set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

#Predict the Test set data
Y_Predict = predict(regressor, newdata = test_set)

#Visualising the training set results
#install.packages("ggplot2")
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
             colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') +
  xlab('Years of Experience') +
  ylab('Salary')

#Visualising the test set results
#install.packages("ggplot2")
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colours = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
            colours = 'blue') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('Years of Experience') +
  ylab('Salary')